### PR TITLE
For #12865, #12990 - Disable swipe to switch tabs gesture when the ke…

### PR DESCRIPTION
**Pull request by person808**
…yboard is visible.

I cannot get my screen recording to work right now, but I've tested on API 21, 24, and 29 (gesture nav and three button nav)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt [hunk 1]
```diff
@@ -309,6 +314,30 @@ class ToolbarGestureHandler(
         }.start()
     }
 
+    private fun isKeyboardVisible(rootView: View): Boolean {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/ToolbarGestureHandler.kt:317]: Maybe we could make this as an extension method off of View?

**Comment by person808** [comment]
Was going to try and write some tests but it seems like the only way to mock the sdk version is by creating an object to wrap `Build.Version.SDK_INT` calls or using reflection to set it. I'm not sure if either of those are things we want to do.

See https://github.com/mockk/mockk/issues/107

**Comment by ekager** [comment]
Do you just want to verify certain things happen on certain API versions? You can use `@Config(sdk = [21])` to annotate your tests with the desired SDK versions to run on with `@RunWith(AndroidJUnit4::class)`. (we do this in AC see if that helps :) )

**Comment by eliserichards** [comment]
Let's get these tests written and then this will be ready for merge 😄

**Pull request merged**